### PR TITLE
True recursive talks data extraction + improved tests

### PR DESCRIPTION
**Pull request by anttijankeri**
New feature to get all html files in the target folder, and all index.html in nested folders within as talks with links and titles. Very overkill and very cool. :)

EDIT:
This PR adds the ability to have more than 1 level of nesting in the talks folder. So you could have a folder like talks/teaching/courseA and talks/teaching/courseB and each could have index.html still being available for data extraction. Essentially more chance to group/organize talks and still have them visible on the main page (or in the slides.json file.)

### CODE REVIEW

File: scripts/extractSlideData.js [hunk 1]
```diff
@@ -35,44 +35,53 @@ const extractSlideData = (folderName) => {
...
         listFilesInDirectory(path.join(talksPath, file), fileNames);
 
-        return fileNames.some((file) => htmlFilter.test(file));
+        return fileNames.forEach((file) => {
+          if (indexFilter.test(file)) {
+            talks.push(
+              file
+                .replace(talksPath, '')
+                .replace(/\\/g, '/')
+                .match(/(?<=\/).+$/)[0]
```
Comment (AhsanAyaz) [inline scripts/extractSlideData.js]: Can you extract this regex out to after line 41? Also, can you add a comment (in the code) about what this does ?

**Review by AhsanAyaz — CHANGES_REQUESTED** [review]
Requesting minor changes. Can you also update the PR's description about what this PR resolves/improves?

File: scripts/extractSlideData.js (hunk 1 above)
Comment (anttijankeri) [reply scripts/extractSlideData.js]: I have instead simplified it now, should be more readable. Can add a comment if necessary.

https://github.com/AhsanAyaz/slides/pull/18/commits/392ef97b0050cefa52069d7e5ead8d89e6a5296e

**Review by AhsanAyaz — APPROVED** [review]
LGTM 👍🏽 Thanks!

**Pull request merged**